work/v1/types.go

```diff
@@ -88,18 +151,46 @@ type SelectivelyOrphan struct {
 	OrphaningRules []OrphaningRule `json:"orphaningRules,omitempty"`
 }
 
-// OrphaningRule identifies a single resource included in this manifestwork
+// ResourceIdentifier identifies a single resource included in this manifestwork
+type ResourceIdentifier struct {
+	// Group is the API Group of the Kubernetes resource.
+	// +kubebuilder:validation:Required
+	// +required
+	Group string `json:"group"`
+
+	// Version is the version of the Kubernetes resource.
+	// +kubebuilder:validation:Required
+	// +required
+	Version string `json:"version"`
```

Comment:
The Version is a property the JSONPath has, right?  The resource itself is identified without a version, but the jsonpath requires the version, but something like WellKnownSTatus would not.

Reply:
I think it is possible that user put the same resource with different versions in the manifests. E.g. crd with v1&v1beta1. Having version here can help to identify which single manifest this option is used to get status. An example

```yaml
apiVersion: work.open-cluster-management.io/v1
Kind: Manifestwork
metadata:
  name: work-1
  namespace: cluster1
spec:
  workload:
    manifests:
    - {resource Foo with version v1}
    - {resource Foo with version v2}
  manifestConfigs:
  - resourceIdentifier:
      resource: Foo
      version: v1
      group: FooGroup
      name: foo
    feedbackRules:
    - type: WellKnownStatus
```

So we only get status from resource Foo with version v1

Reply:
i suppose the version field can be just removed from the struct. we can identify a resource completely w/o the version and the reporting api version should be implicitly aligning w/ the original api version from the manifest. i dont think there's will be cases where we need to e.g. distribute the resource in v1 api but reporting in v1beta1.

Reply:
> I think it is possible that user put the same resource with different versions in the manifests. E.g. crd with v1&v1beta1

oh i didnt know about such use cases before. i though every element in the manifest list should be a logically unique resource

Reply:
> So we only get status from resource Foo with version v1

if the name, namespace, group, and resource are the same, then the distinction between getting v1 status and v1beta1 status is only on whether the JSONPath resolves right?  For the purposes of "WellKnownStatus", you would choose the semantically latest.

For the purposes of locating a resource in a cluster this doesn't provide a distinction.

Reply:
AH, I see. Thanks for the explanation. API updated accordingly.